src/app/_components/Specialists/DistrictFilter.js

```diff
@@ -1,40 +1,39 @@
 'use client';
 
-import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useState } from 'react';
 import { CircularProgress } from '@mui/material';
+import { useSearchParams } from 'next/navigation';
 import { useListDistrict, useSetParam } from '@hooks';
 import { CheckBox } from '@components/CheckBox';
 import { ClearFilterButton, FilterBase } from '@components/Specialists';
-import { useSearchParams } from 'next/navigation';
+import { useDebounceCallback } from '@/app/_hooks';
 
-function DistrictList() {
-  const { data: districts, isLoading } = useListDistrict();
-  const [selectedDistricts, setSelectedDistricts] = useState();
+function DistrictList({ districtsInUrl }) {
+  const districtParam = useSetParam('district');
+  const [selectedDistricts, setSelectedDistricts] = useState(districtsInUrl);
+  const { data: districtList, isLoading } = useListDistrict();
 
-  const searchParams = useSearchParams();
-  const { add, remove } = useSetParam('district');
+  const setParamDebounced = useDebounceCallback(districts => {
+    districtParam.replace(districts);
+  }, 500);
```

Comment:
please extract the timeout value to the variable and use it everywhere. And if case we decide that the debounce for changes in filter has to be other - it will be easy to change

Reply:
thanks, done